modules/uhsg_user_sync/src/SamlAuth/UserSyncSubscriber.php

```diff
@@ -336,42 +336,25 @@ protected function setTechnicalDegreeProgrammes(SamlAuthUserSyncEvent $event, $s
     // Keep track of new technical degree programmes
     $added = 0;
 
-    // Debug Studyright Data
-    if (Settings::get('uhsg_oprek_add_debug_logging', self::UHSG_OPREK_ADD_DEBUG_LOGGING)) {
-      // Loop trough all oodi studyrights
-      $study_rights = $this->oprekService->getStudyRights($student_number);
-
-      foreach ($study_rights as $study_right) {
-        if (!empty($studyright)) {
-          // Debug (a lot of) study right data. Enable only temporarily.
-          \Drupal::logger('uhsg_oprek')->info('setTechnicalDegreeProgrammes(),
-            targeted codes are: <pre>@targeted_codes</pre> and
-            degree_programmes: <pre>@degree_programmes</pre>', [
-            '@targeted_codes' => print_r($study_right->getTargetedCodes(), TRUE),
-            '@degree_programmes' => print_r($known_degree_programme_keys, TRUE),
-          ]);
-        }
-      }
-
-      // Loop trough all sisu studyrights
-      $studyrights = $this->studyRightsService->getActiveStudyRights($student_number);
-      // Loop trough all studyrights
-      foreach($studyrights as $studyright) {
-        if (!empty($studyright)) {
-          // Debug (a lot of) study right data. Enable only temporarily.
-          \Drupal::logger('uhsg_oprek')->info('setTechnicalDegreeProgrammes(),
-            targeted codes are: <pre>@targeted_codes</pre> and
-            degree_programmes: <pre>@degree_programmes</pre>', [
-            '@targeted_codes' => print_r($studyright->getCode(), TRUE),
-            '@degree_programmes' => print_r($known_degree_programme_keys, TRUE),
-          ]);
-        }
-      }
-    }
-
     // Use Oodi Service
     // Map study rights to known degree programmes and create flaggings
     if (!$use_sisu_service && $study_rights = $this->oprekService->getStudyRights($student_number)) {
+      // Debug Studyright Data
```

Comment:
Debug data was kept out of these so that we can debug oodi and sisu data without worrying which service we are using. This way we can see if the data differs, or if there are irregularities in the data.

Reply:
I agree that the aim for drop-in replament and to be able to compare differences in data is a good idea but IMO the data sources and their implementations are already quite different. We can still switch between sisu on doo-oodi ("oprek") with that variable so that is good.

However, the previous way of debugging "both" didnt work, in fact it seems to have caused a problem in such that even if " uhsg_oprek_use_sisu_service = TRUE" it would try to fetch the same data for both oprek and sisu and in many cases an error from oprek would prevent sisu-data to be handled normally (because of the error raised). 

So to be able to compare the responses/data of both integrations it seems best to ensure neither is blocking the execution of the other, so I split the debug handling also. Now "uhsg_oprek_use_sisu_service=TRUE/FALSE" works as one would expect.